libqtile/scripts/migrate.py

```diff
@@ -144,6 +147,47 @@ def f(query, fro=fro, to=to):
     MIGRATIONS.append(f)
 
 
+# Number of migrations that have been removed.
+PREVIOUS_MIGRATIONS = 0
+
+MIGRATION_COUNT_PATTERN = re.compile(r"^migration_count\s*=\s*(\d+)\s*")
+
+
+def get_migration_count(config):
+    # this is fairly horrible, but there's no guarantee we can __import__() a
+    # config file in the current version, since it may not have been migrated
+    # yet. so we parse it manually :)
+    with open(config) as f:
+        for line in f.readlines():
+            match = MIGRATION_COUNT_PATTERN.fullmatch(line)
+            if match is not None:
+                return int(match.group(1))
+
+    # didn't find it, so the file hasn't been migrated before
+    return 0
+
+
+def set_migration_count(config):
+    new_migration_count = "migration_count = {}".format(PREVIOUS_MIGRATIONS + len(MIGRATIONS))
+    written = False
+    with open(config, "r+") as f:
+        lines = f.readlines()
+        f.seek(0)
+        for line in lines:
+            match = MIGRATION_COUNT_PATTERN.fullmatch(line)
+            if match is not None:
+                written = True
+                f.write(new_migration_count)
+                f.write("\n")
+            else:
```

Comment:
What is this doing? Looks like it is prepending the migration count but that's done in the next block after the loop, so I think I'm misunderstanding this

Reply:
There are two cases: 1. the user has an existing `migration_count` that needs to be updated (i.e. the regex matches, but has an old value from the last time we did migrations) which is here. Another case is that a user hasn't used `migrate` before, and `migration_count` doesn't exist in their config file; that's what the case below handles.